### PR TITLE
Feature countries on off

### PR DESCRIPTION
**Pull request by Hkr32**
Ability to disable and enable multiple countries on the list page
feature-countries

### CODE REVIEW

File: upload/admin/controller/localisation/country.php [hunk 1]
```diff
@@ -216,6 +289,7 @@ protected function getList() {
 		$data['sort_name'] = $this->url->link('localisation/country', 'user_token=' . $this->session->data['user_token'] . '&sort=name' . $url);
 		$data['sort_iso_code_2'] = $this->url->link('localisation/country', 'user_token=' . $this->session->data['user_token'] . '&sort=iso_code_2' . $url);
 		$data['sort_iso_code_3'] = $this->url->link('localisation/country', 'user_token=' . $this->session->data['user_token'] . '&sort=iso_code_3' . $url);
+		$data['sort_status'] = $this->url->link('localisation/country', 'user_token=' . $this->session->data['user_token'] . '&sort=status' . $url, true);
```
Comment (straightlight) [inline upload/admin/controller/localisation/country.php:292]: This line won't work in the master branch: https://github.com/opencart/opencart/pull/8225/commits/a33476ab80cc10a7ec609360894fbce9b221e220#diff-ae87ef052a1157e79231def36b367e57R292 .

**Comment by Hkr32** [comment]
Thanks! I have corrected. Now everything is ok?

**Comment by straightlight** [comment]
> @straightlight, Thanks! I have corrected. Now everything is ok?

Not really. Your suggestion is merchant-based but not customer-based nor affiliate-based. Both startup/startup controllers needs to be edited as well since the config_country_id and config_zone_id are still applied from the default store settings. As for the payment and shipping extensions, the geo zones are also configured for each contries and zones. By disabling those options, neither party will be able to identify the root cause when customers will buy the goods from the checkout page.

In addition, disabling a geographic area needs to be informed to customers from the stores (information page). Without relying on payment service providers' API responses, it would then mean that merchants would take full responsibilities to inform customers as opposed to their service providers while, normally, service providers do collect the information by informing the merchants and customers regarding a transaction that occurred.

While there are already extensions on the Opencart.com Marketplace providing the possibility to restrict payment and shipping methods, it would be better to restrict than disable at this point.

**Comment by danielkerr** [comment]
already added

**Pull request closed without merging**